src/components/Table/head/index.js

```diff
@@ -9,26 +9,47 @@ export default function Head(props) {
         columns,
         selectedColumn,
         sortDirection,
+        onColumnSelect,
         resizeColumnDisabled,
         minColumnWidth,
         maxColumnWidth,
+        columnsWidths,
+        onResize,
+        resizeGuideLineHeight,
     } = props;
 
+    const isResizable = columnWidth => !resizeColumnDisabled && columnWidth === undefined;
+
+    const getColumnWidth = (width, defaultWidth, innerWidth) => {
+        if (width === undefined) {
+            if (defaultWidth === undefined) {
+                return innerWidth;
+            }
+            return defaultWidth;
+        }
+        return width;
+    };
+
     if (columns) {
         return columns.map((column, index) => {
-            const { header, sortable, width } = column;
+            const { header, sortable, width, defaultWidth } = column;
             const isSelected = index === selectedColumn;
+            const innerWidth = columnsWidths[index];
             return (
                 <Header
-                    key={uniqueId('header')}
+                    key={`header-${index}`}
```

Comment:
put this in a variable:
```const key = `header-${index}`;```
also in this way you can remove the eslint exception a the top of the file